odbc-api/src/environment.rs

```diff
@@ -271,16 +276,18 @@ impl Environment {
     ///
     /// # Ok::<_, odbc_api::Error>(())
     /// ```
-    pub fn user_data_sources(&mut self) -> Result<Vec<DataSourceInfo>, Error> {
+    pub fn user_data_sources(&self) -> Result<Vec<DataSourceInfo>, Error> {
         self.data_sources_impl(FetchOrientation::FirstUser)
     }
 
     fn data_sources_impl(
-        &mut self,
+        &self,
         direction: FetchOrientation,
     ) -> Result<Vec<DataSourceInfo>, Error> {
         // Find required buffer size to avoid truncation.
 
+        let _lock = self.drivers_lock.lock().expect("cannot lock drivers to query");
```

Comment:
Message in expect is misleading, but unwrap is also fine with me. I do not see how the mutex could be poisend.

Reply:
Alright, I just changed it to `unwrap` for now